### PR TITLE
Bump min supported Python to 3.10

### PR DESCRIPTION
**Pull request by janosh**
following suit with `pymatgen` which in turn follows `numpy`, `scipy` and the rest of the NumFOCUS stack, this PR drops Python 3.9 support.

@utf curious to hear your thoughts. we may want to add 3.12 to CI in the same PR. another open question is whether we want to set `zip`'s new `strict` keyword to `True`. it defaults to `False` which preserves the old behavior of stopping whenever the 1st iterator is exhausted. `True` instead throws an error if all iterators don't have the same length

### CODE REVIEW

**Comment by utf** [comment]
It's already time for a new conda env 😨. Thanks for this @janosh. It makes sense to follow suit with pymatgen.

**Review by utf — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -5,6 +5,7 @@
 [![pypi version](https://img.shields.io/pypi/v/atomate2?color=blue)](https://pypi.org/project/atomate2)
 ![supported python versions](https://img.shields.io/pypi/pyversions/atomate2)
```
Comment (QuantumChemist) [inline README.md:6]: Here is a second python badge still listing 3.9.

Comment (janosh) [reply README.md:6]: that badge pulls the data it displays from PyPI, so will auto-update once this PR is merged and released

Comment (QuantumChemist) [reply README.md:6]: Oki, good to know :)

**Comment by janosh** [comment]
@utf the required CI runs for auto-merging need updating. 3.9 is still required but should be replaced with 3.12

**Pull request merged**